src/main/kotlin/org/rust/lang/core/resolve2/FacadeResolve.kt

```diff
@@ -586,7 +586,7 @@ private fun MacroDefInfo.legacyMacroToPsi(containingScope: RsItemsOwner, info: R
         is DeclMacroDefInfo -> items.legacyMacros.singleOrNull {
             val crate = info.project.crateGraph.findCrateById(crate) ?: return@singleOrNull false
             val defIndex = info.getMacroIndex(it, crate) ?: return@singleOrNull false
-            MacroIndex.equals(defIndex, macroIndex)
+            MacroIndex.equals(defIndex, macroIndex) && it.isEnabledByCfgSelf(crate)
```

Comment:
```suggestion
            // Note that if macro index is equal, the result macro will be cfg-enabled, 
            // since cfg-enabled and cfg-disabled macros have different macro index
            MacroIndex.equals(defIndex, macroIndex)
```